core/opendaq/component/include/opendaq/component_impl.h

```diff
@@ -1114,6 +1120,43 @@ bool ComponentImpl<Intf, Intfs...>::validateComponentId(const std::string& id)
     return id.find(' ') == std::string::npos;
 }
 
+template <class Intf, class... Intfs>
+void ComponentImpl<Intf, Intfs...>::initComponentErrorStateStatus() const
+{
+    // Component error state status is added ("Ok" when a component is created)
+    const auto statusContainerPrivate = this->statusContainer.template asPtr<IComponentStatusContainerPrivate>();
```

Comment:
Call with borrow parameter =True

Reply:
@dejancrn Fixed! However I'm just curious why this is needed or preferred. Does it affect performance? My understanding is that that the variable will go out of scope at the end of `initComponentErrorStateStatus`, thereby decreasing the reference count to the original value.

Reply:
By borrowing you avoid unnecessary reference count increment when smart ptr is created and decrement when destroyed (out of scope). Not super important in configuration part of openDAQ (more important on signal path). However, it makes sense to get a grasp why borrowing is better in cases when you do not keep a reference to object.

Reference count is an atomic variable and as such requires memory barrier and memory bus locks when incremented and decremented.

Reply:
@dejancrn Thank you for the nice explanation! I'll keep this in mind!